node-bootstrapper/app.go

```diff
@@ -100,6 +104,60 @@ func (a *App) provisionStart(ctx context.Context, cse utils.SensitiveString) err
 	return err
 }
 
+// usage example:
+// node-bootstrapper monitor
+func (a *App) Monitor(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 15*time.Minute)
+	defer cancel()
+
+	for {
+		select {
+		case <-timeoutCtx.Done():
+			// If the timeout or cancel occurs, exit with a timeout error
+			return fmt.Errorf("monitoring timed out: %s still active after 15 minutes", BootstrapService)
+		default:
+			_, err := os.Stat(ProvisionJSONFilePath)
+			// Check the active state of the unit
+			_, sysctlerr := a.runSystemctlCommand(ctx, "status", BootstrapService)
+
+			// if service is inactive or failed, error code will be non-zero
+			if sysctlerr == nil || os.IsNotExist(err) {
+				// Unit is still active, sleep for 3 seconds before checking again
+				time.Sleep(3 * time.Second)
+				continue
+			}
+
+			provisionJSON, err := a.getProvisionJSON()
+			if err != nil {
+				return fmt.Errorf("error getting provision.json output, %w", err)
+			}
+			// print to stdout so it will be returned by custom script extension
+			fmt.Println(provisionJSON)
+			return nil
+		}
+	}
+}
+
+// runSystemctlCommand is a generic function that runs a systemctl command with specified arguments
+func (a *App) runSystemctlCommand(ctx context.Context, args ...string) (string, error) {
+	cmd := exec.CommandContext(ctx, "systemctl", args...)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(output)), nil
+}
+
+// getProvisionJSON returns the contents of provision.json containing bootstrap status info
+func (a *App) getProvisionJSON() (string, error) {
+	// Read the file contents
+	data, err := os.ReadFile(ProvisionJSONFilePath)
```

Comment:
FYI I'm thinking about something like adding

```
type IO interface {
  ExecCMD(cmd *exec.Cmd) error
  ReadFile(path string) (os.File, error)
  WriteFile(path string) (os.File, error)
}
```

So, we can mock all the external dependencies and have some unit tests

Reply:
This would be great, though not sure if we will be able to mock the file wait using the new library..